keps/sig-release/2572-release-cadence/README.md

```diff
@@ -0,0 +1,321 @@
+contributors in doing more.
+
+#### SIG Release members
+
+By applying a cadence of 3 releases per year, SIG Release members will gain a
+reduced management overhead. There are also only 3 patch releases to maintain,
+which right now can overlap up to 4. SIG Release will gain more time to ensure
+a seamless transition from the previous release team to the next one. It is
+also possible to include more shadows if the role leads conclude that this is
+appropriate.
+
+### Risks and Mitigations
+
+#### Concentrating risk
+
+In theory a reduced release cadence will cause more changes for every release.
+This means that there will be an increased risk, which would usually be split
+up into 4 dedicated milestones rather than 3.
+
+SIG Release cannot mitigate this risk directly, but is able to track and
+influence it during each release cycle. It's the responsibility of SIG Release,
+together with SIG Testing and SIG Architecture, to identify new gaps and issues
+in the release cadence and mitigate them on a case-by-case basis.
+
+#### Attention to tests
+
+This KEP does not propose any change to the release cycle itself and assumes
+that the same periods for Code and Test Freeze. Assuming that, there is an
+increased risk for flakes and test failures. It will be the responsibility of
+SIG Release to mitigate this, together with the CI signal role. If we speak
+about an overall release cycle enhancement of 3-4 weeks, then we believe that
+SIG Release is able to mitigate this risk over multiple releases.
+
+#### Attention to dependencies
+
+Having fewer releases will introduce the risk of missing dependencies—for
+example, Golang upgrades. This has to be mitigated on a case-by-case basis, in
+the same way as it is being done right now.
+
+## Design Details
+
+### Schedule Policy
+
+The exact details of the release schedule are created by the team leads before
+the actual cycle begins. SIG Release only provides a lightweight policy, which
+is defined as:
+
+1. The first Kubernetes release of a year should start at the second or third
+   week of January to provide people more room after coming back from the
+   Christmas holidays.
+
+2. The last Kubernetes release of a year should be finished by the middle of
+   December.
+
+3. A Kubernetes release cycle has a length of of ~15 weeks.
+
+4. Events like KubeCon will be considered as blocked from development or
+   decision-making. SIG Release will also consider the week before and after
+   the event in the same way.
+
+5. An explicit break of at least two weeks between each release cycle will be
+   enforced.
+
+   This does not mean that zero development can happen during that time.
+   Rather, SIG Release will use this time to do the release retrospective and
+   plan for the next cycle.
+
+### Feedback survey
+
+Each minor Kubernetes release will be an experience survey, which will include
+questions around the release cadence.
```

Comment:
It's not clear to me what changes the collected data is expected to drive. Are you envisioning it informing the endpoints of release plans (like "3 releases per year") or some of the internal milestones (like "how many release candidates")?

A survey on each release that is expected to impact upcoming release schedules seems like it would make schedules less predictable and deterministic. There's also "survey fatigue" to consider, if this is trying to get feedback from the same consumers over and over.

Reply:
Wondering it would make more sense to commit upfront to try to new cadence for X releases and then do a survey? Given the survey fatigue Jordan mentions above and also to give time for people to get used to it (ie change is hard).

Reply:
Yes, I think it makes sense to make an end user survey after 3 releases. We already get feedback from the community around each cycle, but to maintain a broader picture it seems reasonable to focus on end-users. I rephrased the sentence to:

> SIG Release will draft an experience survey after the first three releases from
which the new cadence has been applied. This survey which will include questions
around the release cadence and how it impacted end users.